compiler/qsc_qasm3/src/semantic/lowerer.rs

```diff
@@ -1152,7 +1174,7 @@ impl Lowerer {
         let set_declaration = self.lower_enumerable_set(&stmt.set_declaration);
 
         // Push scope where the loop variable lives.
-        self.symbols.push_scope(ScopeKind::Block);
+        self.symbols.push_scope(ScopeKind::Loop);
```

Comment:
there is no loop scope: https://openqasm.com/language/scope.html#scoping-of-variables